eqcorrscan/utils/plotting.py

```diff
@@ -2184,6 +2184,122 @@ def subspace_fc_plot(detector, stachans, **kwargs):
     return fig
 
 
+@additional_docstring(plotting_kwargs=plotting_kwargs)
+def origintime_detections(detections, method='simple', **kwargs):
+
+    '''
+    Plot detections before and after applying match-filter and visual
+    comparison between them in one day.
+
+    :type detections: list
+    :param detections: list of eqcorrscan.core.match_filter.Detection
+    :type method: str
+    :param method: 'simple' or 'advance' defult to 'simple'
+    {plotting_kwargs}
+
+    :returns: :class:`matplotlib.figure.Figure`
+
+    .. warning::
+        For showing templates' origin, template name must be str of
+        the origin time of each template in match_filter or being changed to
+        origin time later.
+        Something like 2014-10-18T04:02:02
+        If template's name is something else, only detections will be plot.
+
+    .. rubric:: Example
+    >>> from eqcorrscan.core.match_filter import read_detections
+    >>> from eqcorrscan.utils.plotting import origintime_detections
+    >>> # Read text file detections
+    >>> detections = read_detections('path/to/file/filename')
+    >>> origintime_detections(detections, method='advance',
+    ...                       save=Ture, show=True,
+    ...                       savefile='./output.png')
+
+    .. rubric:: Example
+
+    >>> from eqcorrscan.core.match_filter import match_filter
+    >>> detections = match_filter(
+    ...     template_names=['2014-10-18T04:02:02'], template_list=[template],
+    ...     st=stream, threshold=10, threshold_type='MAD', trig_int=3,
+    ...     plot=False, output_cat=False, output_event=True)
+    >>> origintime_detections(detections, method='advance',
+    ...                       save=Ture, show=True,
+    ...                       savefile='./output.png')
+    '''
+    txt = '''
+        Can't show templates!
+        For showing templates' origin, template name must be str of
+        the origin time of each template in all detections.
+        Something like 2014-10-18T04:02:02
+        '''
+    import matplotlib.pyplot as plt
+    day = UTCDateTime(detections[0].detect_time.date)
+    times = [day.datetime, (day+(3600*24)).datetime]
+    plt.figure()
+    plt.plot(times, [0, 0])
+    if method == 'simple':
+        xdetects = [detect.detect_time.datetime for detect in detections]
+        ydetects = np.zeros(len(xdetects))
+        plt.plot(xdetects, ydetects, 'rx', markersize=7, label='child events')
+        try:
+            parents = [detection.template_name for detection in detections]
+            parents = list(set(parents))
+            number_parents = len(parents)
+            parents = [UTCDateTime(parent).datetime for parent in parents]
+            xparents = [parent for parent in parents if
+                        parent.date() == day.date]
+            yparents = np.zeros(len(parents))
+            plt.plot(xparents, yparents, '.', color='blue', label='parents')
+        except Exception as exception:
```

Comment:
Can you be more specific in your caught exception please?  Also, can you wrap only the statement that you expect to generate an expection in a try.. except rather than the whole thing.  It is generally bad practice to wrap with a plain `Exception`, and to wrap all the code.  In general, avoiding `try.. except` and using a well written `if` statement is better as well.

Reply:
Yes, i will write it for ValueError exception.
I am interesting to write it with if statement, but i don't know how can i find a string is proper for UTCDateTime or not yet.